src/hash_value.cpp

```diff
@@ -378,6 +378,22 @@ ValuePtr HashValue::each(Env *env, Block *block) {
     return this;
 }
 
+ValuePtr HashValue::each_key(Env *env, Block *block) {
+    if (!block)
+        return send(env, SymbolValue::intern("enum_for"), { SymbolValue::intern("each_key") });
+
+    keys(env)->send(env, SymbolValue::intern("each"), {}, block);
+    return this;
+}
```

Comment:
What do you think about implementing this in `hash.rb` so we can avoid the creation of the array of keys:

```ruby
def each_key
  each do |key, _|
    yield key
  end
end
```

Reply:
I should say, you can obviously do that in C++ too, but it was just easier for me to illustrate in Ruby quickly. :wink: Really whichever language you want to implement it in...